app/src/main/java/de/westnordost/streetcomplete/quests/address/AddAddressStreet.kt

```diff
@@ -0,0 +1,117 @@
+import de.westnordost.streetcomplete.data.meta.ALL_ROADS
+import de.westnordost.streetcomplete.data.osm.osmquest.OsmElementQuestType
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.mapdata.OverpassMapDataAndGeometryApi
+import de.westnordost.streetcomplete.data.osm.elementgeometry.ElementGeometry
+import de.westnordost.streetcomplete.data.osm.elementgeometry.ElementPolylinesGeometry
+import de.westnordost.streetcomplete.data.tagfilters.FiltersParser
+import de.westnordost.streetcomplete.data.tagfilters.getQuestPrintStatement
+import de.westnordost.streetcomplete.data.tagfilters.toGlobalOverpassBBox
+import de.westnordost.streetcomplete.quests.localized_name.data.RoadNameSuggestionsDao
+import java.util.regex.Pattern
+
+class AddAddressStreet(
+        private val overpassApi: OverpassMapDataAndGeometryApi,
+        private val roadNameSuggestionsDao: RoadNameSuggestionsDao
+) : OsmElementQuestType<AddressStreetAnswer> {
+    override val commitMessage = "Add street/place names to address"
+    override val icon = R.drawable.ic_quest_housenumber_street
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_address_street_title
+
+    override fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String> {
+        val housenumber = tags["addr:housenumber"]
+        return if (housenumber != null) arrayOf(housenumber) else arrayOf()
+    }
+
+    override fun createForm() = AddAddressStreetForm()
+
+    override fun isApplicableTo(element: Element): Boolean? =
+            ADDRESSES_WITHOUT_STREETS_TFE.matches(element)
+
+    override fun download(bbox: BoundingBox, handler: (element: Element, geometry: ElementGeometry?) -> Unit): Boolean {
+        if (!overpassApi.query(getOverpassQuery(bbox), handler)) return false
+        if (!overpassApi.query(getStreetNameSuggestionsOverpassQuery(bbox), this::putRoadNameSuggestion)) return false
+        return true
+    }
+
+    private fun getOverpassQuery(bbox: BoundingBox) =
+            bbox.toGlobalOverpassBBox() + """
+            relation["type"="associatedStreet"];
+            > -> .inStreetRelation;
+
+            $ROADS_WITH_NAMES -> .named_roads;
+
+            $ADDRESSES_WITHOUT_STREETS
+                (around.named_roads:$MAX_DIST_FOR_ROAD_NAME_SUGGESTION) -> .missing_data;
+
+            (.missing_data; - .inStreetRelation;);""" +
+                    getQuestPrintStatement()
+
+    /** return overpass query string to get roads with names around addresses without streets
+     * */
+    private fun getStreetNameSuggestionsOverpassQuery(bbox: BoundingBox) =
+            bbox.toGlobalOverpassBBox() + "\n" + """
+        $ADDRESSES_WITHOUT_STREETS -> .address_missing_street;
+        $ROADS_WITH_NAMES -> .named_roads;
+        way.named_roads(
+            around.address_missing_street: $MAX_DIST_FOR_ROAD_NAME_SUGGESTION);
+        out body geom;""".trimIndent()
```

Comment:
name suggestions now only downloads streets that are around addresses without streets.